### PR TITLE
Added onClick prop to IconButton

### PR DESCRIPTION
**Pull request by DanielvanVliet**
Quick addition to IconButton, I wonder how this was missed as it seems unusable without this prop.

### CODE REVIEW

File: src/components/IconButton/src/index.tsx [hunk 1]
```diff
@@ -35,6 +35,11 @@ export interface IconButtonProps extends BaseProps {
    * `small` is equivalent to the dense button styling.
    */
   size?: 'small' | 'medium';
+
+  /**
+   * Callback fired when the IconButton is clicked.
+   */
+  onClick?: (event: React.ChangeEvent<unknown>) => void;
```
Comment (bddjong) [inline src/components/IconButton/src/index.tsx:42]: Shouldn't the type be a `React.MouseEvent` or something similar

Comment (DanielvanVliet) [reply src/components/IconButton/src/index.tsx]: Yep you're right that's more precise.

**Review by bddjong — APPROVED** [review]

**Review by GewoonMaarten — APPROVED** [review]

**Comment by GewoonMaarten** [comment]
You can also use `onClick?: React.MouseEventHandler<HTMLButtonElement>`

**Comment by bddjong** [comment]
> You can also use `onClick?: React.MouseEventHandler<HTMLButtonElement>;`

We decided on MouseEvent since MouseEventHandler is just an alias which in our opinion doesn't add much

**Comment by DanielvanVliet** [comment]
The `(event: MouseEvent<..>) => void` communicates the usage better / the user doesn't need to lookup what a handler is.

**Pull request merged**